src/components/ComputeModal.js

```diff
@@ -1372,11 +1394,15 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
       value: selectedLeoImage,
       onChange: ({ value }) => {
         const requiresSpark = _.find({ image: value }, leoImages)?.requiresSpark
-        const newSparkMode = requiresSpark ? (sparkMode || 'master') : false
+        const newRuntimeType = Utils.cond(
+          [!!requiresSpark && isDataproc(runtimeType), () => runtimeType],
+          [!!requiresSpark && !isDataproc(runtimeType), () => runtimeTypes.dataprocSingleNode],
```

Comment:
Can't the above 2 lines be simplified to just return runtimeType (without checking isDataproc)?